src/unix/event.rs

```diff
@@ -0,0 +1,354 @@
+        let consumer = unsafe { EventConsumer::from_raw_fd(reader.into_raw_fd()) };
+
+        let cloned_notifier = notifier.try_clone().expect("Failed to clone notifier");
+        let cloned_consumer = consumer.try_clone().expect("Failed to clone consumer");
+
+        cloned_notifier.notify().unwrap();
+        assert!(cloned_consumer.consume().is_ok());
+    }
+
+    #[test]
+    fn test_new_event_notifier_and_consumer() {
+        let (consumer, notifier) = new_event_consumer_and_notifier(EventFlag::empty())
+            .expect("Failed to create notifier and consumer");
+        notifier.notify().unwrap();
+        assert!(consumer.consume().is_ok());
+    }
+
+    #[test]
+    fn test_nonblock() {
+        let (consumer, _notifier) = new_event_consumer_and_notifier(EventFlag::NONBLOCK)
+            .expect("Failed to create notifier and consumer");
+        let r = consumer.consume();
+        match r {
+            Err(ref inner) if inner.kind() == io::ErrorKind::WouldBlock => (),
+            _ => panic!("Unexpected"),
+        }
+    }
+
+    #[test]
+    fn test_cloexec() {
+        let (_consumer, _notifier) = new_event_consumer_and_notifier(EventFlag::CLOEXEC)
+            .expect("Failed to create notifier and consumer");
+
+        // SAFETY: This is safe because we are checking the return value is valid.
+        unsafe {
+            let pid = libc::fork();
+            if pid == 0 {
+                let path = CString::new("/bin/sleep").expect("CString::new failed");
+                let arg0 = CString::new("sleep").expect("CString::new failed");
+                let arg1 = CString::new("5").expect("CString::new failed");
+                let ret = libc::execl(
+                    path.as_ptr(),
+                    arg0.as_ptr(),
+                    arg1.as_ptr(),
+                    std::ptr::null::<i8>(),
+                );
+                assert_eq!(ret, 0, "execl failed");
+            } else {
+                loop {
+                    let output = Command::new("ps")
+                        .args(["-p", pid.to_string().as_str()])
+                        .output()
+                        .expect("Failed to execute ps");
+                    if output
+                        .stdout
+                        .lines()
+                        .any(|line| line.expect("Failed to read line").contains("sleep"))
+                    {
+                        break;
+                    }
+                }
+                let output = Command::new("lsof")
+                    .args(["-p", pid.to_string().as_str()])
+                    .output()
+                    .expect("Failed to execute lsof command");
+                output.stdout.lines().for_each(|line| {
+                    let line = line.expect("Failed to read line");
+                    assert!(!line.contains("PIPE") && !line.contains("eventfd"));
+                });
+            }
```

Comment:
As I mentioned, IMO this is racy also with the new loop.

My suggestion is to have a loop, with the `lsof` inside and try it for a while (e.g. 5 sec like the sleep), I mean something like this in pseudo-code:
```
    let start = Instant::now();
    let timeout = Duration::new(5, 0); // 5 seconds
    let mut passed = false;

    while start.elapsed() < timeout && !passed {
        std::thread::sleep(Duration::from_millis(10));
        
        let output = ...
        ...
           let line = ...
           if line.contain... && ... {
                 passed = true;
           }
     }

     assert!(passed);
```

Reply:
> @germag I thought `lsof` usage was for supporting the test on any `unix`, but honestly I'm not sure if it's available also on BSD/macOS.

true!  

but why not just call `libc::fcntl(..., F_GETFD)` and check if returns `FD_CLOEXEC`?, I mean we don't need to test if the operating system honor `FD_CLOEXEC` we just need to test if the flag is set

Reply:
> > @germag I thought `lsof` usage was for supporting the test on any `unix`, but honestly I'm not sure if it's available also on BSD/macOS.
> 
> true!
> 
> but why not just call `libc::fcntl(..., F_GETFD)` and check if returns `FD_CLOEXEC`?, I mean we don't need to test if the operating system honors `FD_CLOEXEC` we just need to test if the flag is set

This is a great point, @uran0sH WDYT?

Reply:
@uran0sH yes of course, I was talking about the last proposal from @germag:

> but why not just call libc::fcntl(..., F_GETFD) and check if returns FD_CLOEXEC?, I mean we don't need to test if the operating system honors FD_CLOEXEC we just need to test if the flag is set

Reply:
OK, maybe we can just check if FD_CLOEXEC is set, which would simplify this test